File: frontend/app/models/post.js
export default DS.Model.extend({
  title: DS.attr(),
  body: DS.attr(),
  user: DS.belongsTo('user', { async: true }),
  featured: DS.belongsTo('asset-image', { polymorphic: true }),
  images: DS.hasMany('asset-images', { polymorphic: true }),
  audios: DS.hasMany('asset-audios', { polymorphic: true })
});

File: backend/app/models/post.rb
class Post < ActiveRecord::Base
  belongs_to :user
  has_one :featured, -> { where context: 'featured_image' }, as: :assetable, class_name: "Asset::Image", dependent: :destroy
  has_many :images, -> { where context: 'gallery_images' }, as: :assetable, class_name: "Asset::Image", dependent: :destroy
  has_many :audios, -> { where context: 'audios' }, as: :assetable, class_name: "Asset::Audio", dependent: :destroy
end
